raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -516,15 +521,15 @@ public void testEndQuorumEpochRetriesWhileResigned(boolean withKip853Rpc) throws
     }
 
     @ParameterizedTest
-    @ValueSource(booleans = { true, false })
-    public void testResignWillCompleteFetchPurgatory(boolean withKip853Rpc) throws Exception {
+    @EnumSource(value = RaftClientTestContext.RaftProtocol.class, names = {"KIP_595_PROTOCOL", "KIP_853_PROTOCOL"})
```

Comment:
You imported RaftProtocol (`import static ....RaftClientTestContext.RaftProtocol`) so you should be able to reference `RaftProtocol` directly like you do in the method parameter. E.g. `RaftProtocol raftProtocol`.

I am also okay if you want to pull `RaftProtocol` into its own file. E.g. `raft/src/test/java/org/apache/kafka/raft/RaftProtocol.java`

Reply:
IMHO, I would prefer `RaftProtocol` as a single file. 
I will extract it in the next commit.